cg/apps/lims/api.py

```diff
@@ -85,74 +85,114 @@ def _export_artifact(self, lims_artifact):
-                                  if artifact.parent_process.udf.get('Finish Date') is not None]
-                if len(sequence_dates) == 0:
-                    break  # nothing sequenced, break out and return None
-                else:
-                    return min(sequence_dates)
 
-        return None  # if nothing is found just return None
+        step_names_udfs = MASTER_STEPS_UDFS['sequenced_step']
+        sequenced_dates = []
+
+        for process_type in step_names_udfs:
+            artifacts = self.get_artifacts(process_type=process_type, samplelimsid=lims_id)
+
+            for artifact in artifacts:
+                udf_key = step_names_udfs[process_type]
+                if artifact.parent_process and artifact.parent_process.udf.get(udf_key):
+                    sequenced_dates.append((artifact.parent_process.date_run,
+                                            artifact.parent_process.udf.get(udf_key)))
+                if artifact.parent_process and process_type == 'AUTOMATED - NovaSeq Run':
+                    sequenced_dates.append((artifact.parent_process.date_run,
+                                            artifact.parent_process.date_run))
+
+        sorted_dates = sorted(sequenced_dates, key=lambda x: x[0])
+
+        if sorted_dates:
+            log.warning("multiple sequence artifacts found for: %s", lims_id)
+
+        return sorted_dates[-1][1] if sorted_dates else None
 
     def capture_kit(self, lims_id: str) -> str:
         """Get capture kit for a LIMS sample."""
+
+        step_names_udfs = MASTER_STEPS_UDFS['capture_kit_step']
+        capture_kits = set()
+
         lims_sample = Sample(self, id=lims_id)
         capture_kit = lims_sample.udf.get('Capture Library version')
         if capture_kit and capture_kit != 'NA':
             return capture_kit
         else:
-            artifacts = self.get_artifacts(
-                samplelimsid=lims_id,
-                process_type='CG002 - Hybridize Library  (SS XT)',
-                type='Analyte'
-            )
-            udf_key = 'SureSelect capture library/libraries used'
-            capture_kits = set(artifact.parent_process.udf.get(udf_key) for artifact in artifacts)
-            if len(capture_kits) == 1:
-                return capture_kits.pop()
-            else:
-                message = f"Capture kit error: {lims_sample.id} | {capture_kits}"
-                raise LimsDataError(message)
+            for process_type in step_names_udfs:
+                artifacts = self.get_artifacts(
+                    samplelimsid=lims_id,
+                    process_type=process_type,
+                    type='Analyte'
+                )
+                udf_key = step_names_udfs[process_type]
+                capture_kits = capture_kits.union(set(artifact.parent_process.udf.get(udf_key) for
+                                                      artifact in artifacts))
+                if len(capture_kits) == 1:
+                    continue
```

Comment:
previously we returned this one item, now I get the feeling that we skip over it in the for loop, right? intentional?

Reply:
Going to solve this later today